raiden/transfer/channel.py

```diff
@@ -1220,14 +1255,29 @@ def register_secret_endstate(
         )
 
 
+def register_onchain_secret_endstate(
+        end_state: NettingChannelEndState,
+        secret: typing.Secret,
+        secrethash: typing.SecretHash,
+) -> None:
+    if is_lock_locked(end_state, secrethash):
+        pendinglock = end_state.secrethashes_to_lockedlocks[secrethash]
+        del end_state.secrethashes_to_lockedlocks[secrethash]
```

Comment:
I think it is better to use `_del_lock` . At this point it is possible to have the `secret` in `secrethashes_to_unlockedlocks` in the case where the on-chain secret reveal transaction was sent by this node during the unsafe period.

Reply:
Good point, but then it wouldn't be in `secrethashes_to_lockedlocks` any more, right?

Reply:
Correct. So it can either be in `secrethashes_to_lockedlocks` or `secrethashes_to_unlockedlocks`.

Reply:
We missed that `is_lock_locked` only checks in  `secrethashes_to_lockedlocks`, so should be fine as it was before.

```py
secrethashes_to_lockedlocks
```

Reply:
My point was that locks can either be in `secrethashes_to_lockedlocks` , `secrethashes_to_unlockedlocks` or `secrethashes_to_onchain_unlockedlocks`. 

If you add it to `secrethashes_to_onchain_unlockedlocks` with `register_onchain_secret_endstate` , the lock could have been in either `secrethashes_to_lockedlocks` or `secrethashes_to_unlockedlocks` - we need to try to remove it from both. 

So `if is_lock_locked(end_state, secrethash):`  should be removed in my opinion.